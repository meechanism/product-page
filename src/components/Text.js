import styled from 'styled-components';
import Colors from '../styles/colors';

export default styled.p`
  font-size: 1rem;
  line-height: 1.5rem;
  margin: 0 0 1rem;
  color: ${(props) => (props.muted ? Colors.gray : Colors.black)};
`;
