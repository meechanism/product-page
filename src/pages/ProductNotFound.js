import React from 'react';
import { Link } from 'react-router-dom';

import PageWrapper from '../components/PageWrapper';
import Text from '../components/Text';

const ProductNotFound = () => {
  return (
    <PageWrapper>
      <h1>Oops!</h1>
      <Text>
        Sorry, we couldn't find that product. Please contact customer support if
        you need assistance.
      </Text>
      <Text>
        In the meantime, you can <Link to="/">continue shopping</Link>.
      </Text>
    </PageWrapper>
  );
};

export default ProductNotFound;
