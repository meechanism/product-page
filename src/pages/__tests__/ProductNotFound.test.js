import React from 'react';

import ProductNotFound from '../ProductNotFound';
import Text from '../../components/Text';

describe('ProductNotFound page', () => {
  it('should render', () => {
    const output = shallow(<ProductNotFound />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should contain header', () => {
    const output = shallow(<ProductNotFound />);
    expect(output.find('h1').length).toBe(1);
    expect(output.find('h1').text()).toBe('Oops!');
  });

  it('should contain two supporting  texts', () => {
    const output = shallow(<ProductNotFound />);
    expect(output.find(Text).length).toBe(2);
  });
});
