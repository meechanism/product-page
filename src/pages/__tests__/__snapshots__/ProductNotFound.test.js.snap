// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductNotFound page should render 1`] = `
<_default>
  <h1>
    Oops!
  </h1>
  <styled.p>
    Sorry, we couldn't find that product. Please contact customer support if you need assistance.
  </styled.p>
  <styled.p>
    In the meantime, you can 
    <Link
      to="/"
    >
      continue shopping
    </Link>
    .
  </styled.p>
</_default>
`;
